presto-main/src/main/java/com/facebook/presto/execution/scheduler/FixedSourcePartitionedScheduler.java

```diff
@@ -67,14 +78,22 @@ public ScheduleResult schedule()
             scheduledTasks = true;
         }
 
-        ScheduleResult schedule = sourcePartitionedScheduler.schedule();
-        return new ScheduleResult(schedule.isFinished(), newTasks, schedule.getBlocked());
+        CompletableFuture<?> blocked = CompletableFuture.completedFuture(null);
+        while (!sourcePartitionedSchedulers.isEmpty()) {
+            ScheduleResult schedule = sourcePartitionedSchedulers.get(0).schedule();
+            blocked = schedule.getBlocked();
+            if (!blocked.isDone() || !schedule.isFinished()) {
```

Comment:
I don't understand the logic here.

I assume `schedule.isFinished` implies `blocked.isDone`. Let me know if this assumption is wrong.

Say there are two sources, does this mean we never schedule the second source until scheduling of first source is completely finished?